Passing config to CLI causes assertion error

When using the CLI and `--config` an assertion error is generated if you call `sass-lint` from a directory that doesn't include a `.sass-lint.yml` file or a `package.json` with a config file specified within.

```bash
dan@Dans-Box workspace $ sass-lint --config ~/workspace/sass-lint/.sass-lint.yml ~/vagrant/www/vhosts/test-proj.dev/public/app/scss/main.scss

assert.js:86
  throw new assert.AssertionError({
        ^
AssertionError: missing path
    at Module.require (module.js:363:3)
    at require (module.js:384:17)
    at module.exports (/usr/local/lib/node_modules/sass-lint/lib/config.js:45:12)
    at Function.sassLint.getConfig (/usr/local/lib/node_modules/sass-lint/index.js:18:10)
    at Function.sassLint.format (/usr/local/lib/node_modules/sass-lint/index.js:89:21)
    at detectPattern (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:17:20)
    at /usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:58:5
    at Array.forEach (native)
    at Object.<anonymous> (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:57:16)
    at Module._compile (module.js:460:26)
```

Have a working fix ready to go, will just have a look at adding some tests

--- a/bin/sass-lint.js
+++ b/bin/sass-lint.js
@@ -14,11 +14,11 @@ var detectPattern = function (pattern) {
       formatted;
 
   detects = lint.lintFiles(pattern, configOptions, configPath);
-  formatted = lint.format(detects);
+  formatted = lint.format(detects, configOptions, configPath);
 
 
   if (program.verbose) {
-    lint.outputResults(formatted);
+    lint.outputResults(formatted, configOptions, configPath);
   }
 
 

--- a/tests/cli.js
+++ b/tests/cli.js
@@ -113,7 +113,7 @@ describe('cli', function () {
           return done();
         }
         else {
-          return done(new Error('warnings/errors were returned'));
+          return done(new Error('warnings/errors were expected to be returned but weren\'t'));
         }
       }
     });

--- a/tests/cli.js
+++ b/tests/cli.js
@@ -33,4 +33,89 @@ describe('cli', function () {
     });
 
   });
+
+  // Test custom config path
+
+  it('should return JSON from a custom config', function (done) {
+    var command = 'sass-lint -c tests/yml/.json-lint.yml tests/sass/cli.scss --verbose';
+
+    childProcess.exec(command, function (err, stdout) {
+
+      if (err) {
+        return done(err);
+      }
+      else {
+        try {
+          JSON.parse(stdout);
+          return done();
+        }
+        catch (e) {
+          return done(new Error('Not JSON'));
+        }
+      }
+    });
+  });
+
+  // Test 0 errors/warnings when rules set to 0 in config
+
+  it('should return no errors/warnings', function (done) {
+    var command = 'sass-lint -c tests/yml/.json-lint.yml tests/sass/cli.scss --verbose';
+
+    childProcess.exec(command, function (err, stdout) {
+
+      var result = '';
+
+      if (err) {
+        return done(err);
+      }
+
+      else {
+        try {
+          result = JSON.parse(stdout);
+        }
+        catch (e) {
+          return done(new Error('Not JSON'));
+        }
+
+        if (result[0].warningCount === 0 && result[0].errorCount === 0) {
+          return done();
+        }
+        else {
+          return done(new Error('warnings/errors were returned'));
+        }
+
+      }
+    });
+  });
+
+  // Test 1 warning when rules set to 0 in config
+
+  it('should return a warning', function (done) {
+    var command = 'sass-lint -c tests/yml/.color-keyword-errors.yml tests/sass/cli.scss --verbose';
+
+    childProcess.exec(command, function (err, stdout) {
+
+      var result = '';
+
+      if (err) {
+        return done(err);
+      }
+
+      else {
+        try {
+          result = JSON.parse(stdout);
+        }
+        catch (e) {
+          return done(new Error('Not JSON'));
+        }
+
+        if (result[0].warningCount === 1 && result[0].errorCount === 0) {
+          return done();
+        }
+        else {
+          return done(new Error('warnings/errors were returned'));
+        }
+      }
+    });
+  });
 });

--- a/tests/sass/cli.scss
+++ b/tests/sass/cli.scss
@@ -1,0 +1,3 @@
+.cli {
+  color: red;
+}

--- a/tests/yml/.color-keyword-errors.yml
+++ b/tests/yml/.color-keyword-errors.yml
@@ -1,0 +1,58 @@
+options:
+  formatter: json
+files:
+  include: '**/*.s+(a|c)ss'
+rules:
+  # Extends
+  extends-before-mixins: 0
+  extends-before-declarations: 0
+  placeholder-in-extend: 0
+
+  # Mixins
+  mixins-before-declarations: 0
+
+  # Line Spacing
+  one-declaration-per-line: 0
+  empty-line-between-blocks: 0
+  single-line-per-selector: 0
+
+  # Disallows
+  no-debug: 0
+  no-duplicate-properties: 0
+  no-empty-rulesets: 0
+  no-extends: 0
+  no-ids: 0
+  no-important: 0
+  no-warn: 0
+  no-color-keywords: 1
+  no-invalid-hex: 0
+  no-css-comments: 0
+  no-color-literals: 0
+  no-vendor-prefix: 0
+
+  # Style Guide
+  border-zero: 0
+  clean-import-paths: 0
+  empty-args: 0
+  hex-length: 0
+  hex-notation: 0
+  indentation: 0
+  leading-zero: 0
+  nesting-depth: 0
+  property-sort-order: 0
+  quotes: 0
+  variable-for-property: 0
+  zero-unit: 0
+
+  # Inner Spacing
+  space-after-comma: 0
+  space-before-colon: 0
+  space-after-colon: 0
+  space-before-brace: 0
+  space-before-bang: 0
+  space-after-bang: 0
+  space-between-parens: 0
+
+  # Final Items
+  trailing-semicolon: 0
+  final-newline: 0

--- a/tests/yml/.json-lint.yml
+++ b/tests/yml/.json-lint.yml
@@ -1,0 +1,58 @@
+options:
+  formatter: json
+files:
+  include: '**/*.s+(a|c)ss'
+rules:
+  # Extends
+  extends-before-mixins: 0
+  extends-before-declarations: 0
+  placeholder-in-extend: 0
+
+  # Mixins
+  mixins-before-declarations: 0
+
+  # Line Spacing
+  one-declaration-per-line: 0
+  empty-line-between-blocks: 0
+  single-line-per-selector: 0
+
+  # Disallows
+  no-debug: 0
+  no-duplicate-properties: 0
+  no-empty-rulesets: 0
+  no-extends: 0
+  no-ids: 0
+  no-important: 0
+  no-warn: 0
+  no-color-keywords: 0
+  no-invalid-hex: 0
+  no-css-comments: 0
+  no-color-literals: 0
+  no-vendor-prefix: 0
+
+  # Style Guide
+  border-zero: 0
+  clean-import-paths: 0
+  empty-args: 0
+  hex-length: 0
+  hex-notation: 0
+  indentation: 0
+  leading-zero: 0
+  nesting-depth: 0
+  property-sort-order: 0
+  quotes: 0
+  variable-for-property: 0
+  zero-unit: 0
+
+  # Inner Spacing
+  space-after-comma: 0
+  space-before-colon: 0
+  space-after-colon: 0
+  space-before-brace: 0
+  space-before-bang: 0
+  space-after-bang: 0
+  space-between-parens: 0
+
+  # Final Items
+  trailing-semicolon: 0
+  final-newline: 0

